### PR TITLE
Start `inBackupHealthReport` during migrations

### PR DESCRIPTION
**Pull request by george-dorin**
- Rename the `InBackupHealthReport` to `StartUpHealthReport`
- Start the `StartUpHealthReport` in the `handleNodeVersioning` method so it would cover both db migrations and backups
- Stop using the `DefaultServeMux` to avoid issues when testing migrations and rollbacks

### CODE REVIEW

File: core/services/health.go [hunk 1]
```diff
@@ -50,7 +51,7 @@ func (i *InBackupHealthReport) Stop() {
 
 func (i *InBackupHealthReport) Start() {
```
Comment (george-dorin) [inline core/services/health.go]: Maybe change the name of `InBackupHealthReport`?

Comment (jmank88) [reply core/services/health.go]: Maybe something with `Init`, `Boot`, `StartUp`...?

File: .changeset/soft-maps-ring.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"chainlink": patch
+---
+
+#changed During database migrations the `InBackupHealthReport` will be started.
```
Comment (jmank88) [inline .changeset/soft-maps-ring.md]: Maybe rephrase to be from the node ops perspective? Something about health being reported all throughout start-up, without delay?

File: core/store/migrate/migrate.go [hunk 3]
```diff
@@ -99,12 +101,15 @@ func ensureMigrated(ctx context.Context, db *sql.DB) error {
...
-func Migrate(ctx context.Context, db *sql.DB) error {
+func Migrate(ctx context.Context, db *sql.DB, healthReportPort uint16, lggr logger.Logger) error {
 	if err := ensureMigrated(ctx, db); err != nil {
 		return err
 	}
 	// WithAllowMissing is necessary when upgrading from 0.10.14 since it
 	// includes out-of-order migrations
+	ibhr := services.NewInBackupHealthReport(healthReportPort, lggr)
+	ibhr.Start()
+	defer ibhr.Stop()
```
Comment (jmank88) [inline core/store/migrate/migrate.go]: Can we do this from the caller instead? Otherwise `ensureMigrated` is not covered.

Comment (jmank88) [reply core/store/migrate/migrate.go]: Actually, maybe skip several callers. I think it is important to include as much of the startup time as possible, plus this will avoid the leaking in to the administrative commands that don't need to be reporting health in the first place, like `ResetDatabase`.

File: core/cmd/shell.go
Comment (jmank88) [inline core/cmd/shell.go:1]: Is just `handleNodeVersioning` going to be enough? How much time passes between the end of this call and the start of the normal health report?

**Review by jmank88 — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]
